### PR TITLE
Add 'filterNot' to Maybe, NEL and List [ #78 ]

### PR DESCRIPTION
**Pull request by tomecko**

### CODE REVIEW

File: docs/NEL.md [hunk 1]
```diff
@@ -139,6 +139,14 @@ NEL[A].filter(fn: (element: A) => Boolean): List[A]
 
 Returns a new list, keeping only elements for which the predicate returns true.
 
+### filterNot
+
+```scala
+NEL[A].filterNot(fn: (element: A) => Boolean): List[A]
```
Comment (ulfryk) [inline docs/NEL.md]: `A => Boolean`

File: test/nel-spec.js [hunk 2]
```diff
@@ -107,6 +107,13 @@ describe('A Non-Empty immutable list', function () {
 
     })
 
+    it('will be filtered with filterNot', function () {
+        var n1 = NEL.fromList(List.fromArray([1, 2, 3, 4, 5, 6, 7, 8])).some()
+        expect(n1.filterNot(function (a) {
+            return a % 2 == 0
```
Comment (ulfryk) [inline test/nel-spec.js]: `==` -> `===` (also mabe fix this in `filter` tests)

File: docs/NEL.md (hunk 1 above)
Comment (tomecko) [reply docs/NEL.md]: This happens in many places. I can change it but I'd rather it in a separate pull request for clarity.

File: test/nel-spec.js (hunk 2 above)
Comment (tomecko) [reply test/nel-spec.js]: This happens in many places. I can change it but I'd rather do it in a separate pull request for clarity.

File: docs/NEL.md (hunk 1 above)
Comment (ulfryk) [reply docs/NEL.md]: I mean `.filterNot(fn: A => Boolean)`, but if there is more `(x: A) => B` then :+1: for separate PR.

**Pull request merged**